lib/validators/specResolver.js

```diff
@@ -488,6 +501,281 @@ class SpecResolver {
     }
     return Promise.resolve(self);
   }
+
+  /**
+ * Deletes all the references to allOf from all the model definitions in the swagger spec.
+ */
```

Comment:
```javascript
/**
 * Resolves the discriminator by replacing all the references to the parent model with a oneOf array containing
 * references to the parent model and all its child models. It also modifies the discriminator property in
 * the child models by making it a constant (enum with one value) with the value expected for that model 
 * type on the wire. 
 * For example: There is a model named "Animal" with a discriminator as "animalType". Models like "Cat", "Dog",
 * Shark" are children (having "allof": [ { "$ref": "#/definitions/Animal" } ] on) of "Animal" in the swagger spec. 
 *
 * - This method will replace all the locations in the swagger spec that have a reference to the
 * parent model "Animal" ("$ref": "#/definitions/Animal") except the allOf reference with a oneOf reference 
 * "oneOf": [ { "$ref": "#/definitions/Animal" }, { "$ref": "#/definitions/Cat" },  { "$ref": "#/definitions/Dog" }, { "$ref": "#/definitions/Tiger" }  ]
 *
 * - It will also add a constant value (name of that animal on the wire or the value provided by 
 * "x-ms-discriminator-value") to the discrimiantor property "animalType" for each of the child models. 
 * For example:  the Cat model's discriminator property will look like:
 * "Cat": { "required": [ "animalType" ], "properties": { "animalType": { "type": "string", "enum": [ "Cat" ] },  . . } }.
 */
```

Reply:
👍